### ISSUE TITLE
Cannot click multiple checkbox if using same v-model

### ISSUE THREAD

**Issue by yeungc** [bootstrap-vue/bootstrap-vue#431]
Hello, I just found a problem, if I use same v-model for multiple b-form-checkbox, only one of it can be clicked, please see example:
https://jsfiddle.net/s0Lafhrk/

But if I use native HTML code instead of b-form-checkbox, the problem will be solved:
https://jsfiddle.net/yth2z0r3/

May I know how to fix it? Thanks.

**Comment by tmorehouse** [comment]
I wonder if this is because of how the value is bound to the checkbox inside the component:

```html
<input
       type="checkbox"
       :id="_id"
       :name="name"
       :value="value"
       :disabled="disabled"
       :class="[custom?'custom-control-input':null]"
       :checked="checked===value"
       @change="$emit('change',$event.target.checked?value:uncheckedValue)"
>
```

It might need to be changed to:

```html
<input
       type="checkbox"
       :id="_id"
       :name="name"
       :value="value"
       v-model="value"
       :disabled="disabled"
       :class="[custom?'custom-control-input':null]"
       :checked="checked===value"
       @change="$emit('change',$event.target.checked?value:uncheckedValue)"
>
```

But not sure if that would affect how the `uncheckedValue` is applied

**Comment by tmorehouse** [comment]
The other option might be to create a composite component that makes each individual `b-form-checkbox` value into an array result.

**Comment by alexsasharegan** [comment]
### Vue Examples
[Basic Checkbox](https://vuejs.org/v2/guide/forms.html#Checkbox-1)
[Component implements v-model](https://vuejs.org/v2/guide/components.html#Customizing-Component-v-model)

**Comment by alexsasharegan** [comment]
So I guess checkboxes are implemented as arrays under the hood. Check this craziness out:

https://jsfiddle.net/s0Lafhrk/3/

**Comment by alexsasharegan** [comment]
Still some kinks to work out when unchecking the box.

**Comment by alexsasharegan** [comment]
This works fairly well: https://jsfiddle.net/s0Lafhrk/4/

**Comment by alexsasharegan** [comment]
Yeah, that appears to have full feature parity with the native Vue example @yeungc provided (https://jsfiddle.net/yth2z0r3/)

**Comment by alexsasharegan** [comment]
So the question is, how does vue support v-model being any type?

**Comment by alexsasharegan** [comment]
Alright, this will support the old style, and the array style v-model: https://jsfiddle.net/s0Lafhrk/5/

**Comment by yeungc** [comment]
Oh good, thank you so much Alex!

**Comment by pkat** [comment]
Thanks! I just ran into this scenario today!

**Comment by alexsasharegan** [comment]
@pkat Thanks for checking the issue list first! I believe you'll see this fix in v0.16.2

**Comment by alexsasharegan** [comment]
@yeungc & @pkat If you guys are enjoying bootstrap-vue, let us know with a ⭐️ ! We appreciate your feedback!

**Comment by yeungc** [comment]
@alexsasharegan Sure no problem :)

**Comment by playwolf719** [comment]
> @pkat Thanks for checking the issue list first! I believe you'll see this fix in v0.16.2

h， this is still a problem in bootstrap 2.0.0-rc.11，b-form-checkbox-group could not been check for multi options when I put it into a component.

**Comment by tmorehouse** [comment]
@playwolf719 please open a new issue, and include some example code that can reproduce the problem.
